File: patterns/watering-system/rules-python/watering_system_trigger.py
from openhab import rule, Registry
from openhab.triggers import ItemStateChangeTrigger


@rule(triggers=[ItemStateChangeTrigger("SomeCondition", state="ON")])
class StartZone1Watering:
    def execute(self, module, input):
        Registry.getItem("VT_Watering_Zone1").postUpdate("START")

File: patterns/watering-system/rules-js/watering-system-trigger.js
rules.JSRule({
  name: "Start Zone 1 Watering",
  triggers: [triggers.ItemStateChangeTrigger("SomeCondition", undefined, "ON")],
  execute: (event) => {
    items.getItem("VT_Watering_Zone1").postUpdate("START");
  }
});
